### PR TITLE
Use osgi webconsole REST to stop swookiee

### PR DESCRIPTION
**Pull request by dhiller**

### CODE REVIEW

File: common/swookiee [hunk 1]
```diff
@@ -69,21 +69,43 @@ clean_start() {
...
-  if [ $RES -eq 0 ]; then
-    echo "OK"
+  # use webconsole to stop osgi system bundle
+  echo "Stopping swookiee system bundle"
+  curl -X POST -s -f -u "${OSGI_WEBCONSOLE_USER:-admin}:${OSGI_WEBCONSOLE_PWD:-admin123}" \
+                  "http://localhost:${OSGI_WEBCONSOLE_PORT:-18080}/system/console/bundles/0?action=stop" \
+                  > /dev/null
+  if [ "$?" -ne 0 ]; then
+    echo "Failed to stop swookiee system bundle!"
+    exit 1
```
Comment (thgreiner) [inline common/swookiee]: We should not exit here. Suppose the process is a state where it does not accept network connections anymore - the stop script should still kill it forcefully.

File: common/swookiee [hunk 2]
```diff
@@ -69,21 +69,43 @@ clean_start() {
...
+  fi
+
+  # wait for swookiee to finish clean shutdown
+  n=3
```
Comment (thgreiner) [inline common/swookiee]: Suggest to use n=30 and "sleep 1".

File: common/swookiee [hunk 3]
```diff
@@ -69,21 +69,43 @@ clean_start() {
...
+  until [ $n -le 0 ]
```
Comment (dguenms) [inline common/swookiee]: In accordance with the existing code style, I would put `until [  ]; do` in one line.

**Review by dguenms — COMMENTED** [review]
Waiting for Lars's opinion here, but I like the approach.

File: common/swookiee [hunk 4]
```diff
@@ -69,21 +69,43 @@ clean_start() {
...
-  # Kills the swookiee osgi process
-  echo "Stopping swookiee"
-  kill -15 $pid
-  RES=$?
-  if [ $RES -eq 0 ]; then
-    echo "OK"
+  # use webconsole to stop osgi system bundle
+  echo "Stopping swookiee system bundle"
+  curl -X POST -s -f -u "${OSGI_WEBCONSOLE_USER:-admin}:${OSGI_WEBCONSOLE_PWD:-admin123}" \
+                  "http://localhost:${OSGI_WEBCONSOLE_PORT:-18080}/system/console/bundles/0?action=stop" \
```
Comment (thgreiner) [inline common/swookiee]: Why port 18080 as default? Shouldn't it be 8080?

File: common/swookiee [hunk 5]
```diff
@@ -69,21 +69,43 @@ clean_start() {
...
+    n=$((n-1))
+    echo "Waiting for swookiee to shutdown, $n retries left"
+    sleep 10
+  done
+
+  pid=$(check_status)
+  if [ "$pid" ] ; then
+    # Kill the swookiee osgi process
+    echo "Stopping swookiee cleanly was not successful - killing swookiee process"
+    kill -9 "$pid"
```
Comment (thgreiner) [inline common/swookiee]: Do we really want to use 'kill -9' directly here? I suggest to first use a kill -15 and another grace period of 5 to 10 seconds - only then 'kill -9'.

**Pull request closed without merging**